### PR TITLE
Expose Cached Credential Service request ID in tokenResponse

### PR DESCRIPTION
**Pull request by logdog16**
Following up on PR #1866 to expose this data in a way MSAL C++ can access and add to our telemetry.

ADO Item: https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2391983

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java
Comment (iamgusain) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java:1]: LGTM

Let's add this change behind a flight, so we have more control.
You can refer to this [PR](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1961) to see the pattern for how to enable flights in common code.
or let me know if you need any help with this.

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java [hunk 1]
```diff
@@ -619,10 +620,26 @@ protected TokenResult getTokenResultFromHttpResponse(@NonNull final HttpResponse
...
+                    mapWithAdditionalEntry.put(XMS_CCS_REQUEST_ID, ccsRequestId);
+                    if(null != tokenResponse.getExtraParameters())
+                    {
+                        for(Map.Entry<String, String> entry : tokenResponse.getExtraParameters())
+                        {
+                            mapWithAdditionalEntry.put(entry.getKey(), entry.getValue());
+                        }
+                    }
+                    tokenResponse.setExtraParameters(mapWithAdditionalEntry.entrySet());
+                }
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java]: The formatting / placement of the curly braces doesn't match with the format used in the rest of the codebase

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java [hunk 2]
```diff
@@ -619,10 +620,26 @@ protected TokenResult getTokenResultFromHttpResponse(@NonNull final HttpResponse
                     tokenResponse.setCliTelemSubErrorCode(cliTelemInfo.getServerSubErrorCode());
                 }
             }
+            final String ccsRequestId = response.getHeaderValue(XMS_CCS_REQUEST_ID, 0);
+            if(null != ccsRequestId)
+            {
+                if(null != tokenResponse)
+                {
+                    Map<String, String> mapWithAdditionalEntry = new HashMap<String, String>();
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java]:
```suggestion
                    final Map<String, String> mapWithAdditionalEntry = new HashMap<String, String>();
```

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java [hunk 3]
```diff
@@ -619,10 +620,26 @@ protected TokenResult getTokenResultFromHttpResponse(@NonNull final HttpResponse
...
+                    mapWithAdditionalEntry.put(XMS_CCS_REQUEST_ID, ccsRequestId);
+                    if(null != tokenResponse.getExtraParameters())
+                    {
+                        for(Map.Entry<String, String> entry : tokenResponse.getExtraParameters())
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java]:
```suggestion
                        for(final Map.Entry<String, String> entry : tokenResponse.getExtraParameters())
```

**Review by shahzaibj — APPROVED** [review]

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java (hunk 2 above)
Comment (logdog16) [reply common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java]: Fixed in latest commit

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java (hunk 3 above)
Comment (logdog16) [reply common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java]: Fixed in latest commit

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java (hunk 1 above)
Comment (logdog16) [reply common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java]: Thanks for catching this! I should have been more careful to look at existing style when making my changes, I just let muscle memory from working in C++ libs take over!

Fixed to use opening bracket on originating line of the new scope in latest commit

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java
Comment (logdog16) [reply common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java:1]: Alright! The change is now behind a (default on) flight

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java [hunk 4]
```diff
@@ -619,10 +622,23 @@ protected TokenResult getTokenResultFromHttpResponse(@NonNull final HttpResponse
                     tokenResponse.setCliTelemSubErrorCode(cliTelemInfo.getServerSubErrorCode());
                 }
             }
-
-            SpanExtension.current().setAttribute(
-                    AttributeName.ccs_request_id.name(),
-                    response.getHeaderValue(XMS_CCS_REQUEST_ID, 0));
+            final String ccsRequestId = response.getHeaderValue(XMS_CCS_REQUEST_ID, 0);
+            if(null != ccsRequestId){
```
Comment (iamgusain) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java]: nit: space after if

**Review by iamgusain — APPROVED** [review]

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java (hunk 4 above)
Comment (logdog16) [reply common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java]: Thanks for catching this, I realized the style I was using wasn't in sync with the rest of the code (I wasn't using a space between if/for and the opening parens), so I fixed all the instances I wrote in my latest rev.

**Pull request merged**